test/kv679_eqc.erl

```diff
@@ -0,0 +1,493 @@
+                                                       UseLastGet == false ->
+    new_object(Key, Value);
+generate_put_obj(_Key, Value, LastGet, true) ->
+    O = riak_object:update_value(LastGet, Value),
+    O1= riak_object:update_last_modified(O),
+    riak_object:apply_updates(O1).
+
+new_object(K, V) ->
+    riak_object:new(?BUCKET, K, V).
+
+%% this is a cut down and (vastly!) simplified version of what riak
+%% does when coordinating a put
+coord_put(VId, Epoch, _LocalObj=error, IncomingObject) ->
+    StartTime = timestamp(),
+    Epoch2 = Epoch +1,
+    EpochActor = epoch_actor(VId, Epoch2),
+    Obj = riak_object:increment_vclock(IncomingObject, EpochActor, StartTime),
+    {ok, Dot} = vclock:get_dot(EpochActor, riak_object:vclock(Obj)),
+    {Epoch2, Obj, Dot};
+coord_put(VId, Epoch, {ok, LocalObj}, IncomingObject) ->
+    StartTime = timestamp(),
+    {Epoch2, Actor} = case highest_epoch(VId, LocalObj) of
+                          0 -> %% i.e. never acted on local
+                              {_A2, E2}=A = epoch_actor(VId, Epoch+1),
+                              {E2, A};
+                          Acted when is_integer(Acted) ->
+                              maybe_new_epoch_actor(VId, Epoch, LocalObj, IncomingObject)
+                      end,
+    Obj = riak_object:update(false, LocalObj, IncomingObject, Actor, StartTime),
+    {ok, Dot} = vclock:get_dot(Actor, riak_object:vclock(Obj)),
+    {Epoch2, Obj, Dot}.
+
+%% this is a cut down and (vastly!) simplified version of what riak
+%% does when receiving a non-coorindating (replica, handoff, mdc, read
+%% repair etc) put
+replica_put(VId, Epoch, error, IncomingObj) ->
+    %% NOTE we _may_ update the vclock here!
+    case highest_epoch(VId, IncomingObj) of
+        0 ->
+            %% In this model all actors have epochs of at least 1, in
+            %% this case no epoch means no actor, so nowt funny here
+            {Epoch, IncomingObj};
+        _N ->
+            %% wut? Locally there is no object, but remotely there is
+            %% an object with this actor in, that's broken
+            E2 = Epoch + 1,
+            Actor = epoch_actor(VId, E2),
+            {E2, riak_object:new_actor_epoch(IncomingObj, Actor)}
+    end;
+replica_put(_VId, Epoch, {ok, LocalObj}, IncomingObject) ->
+    {Epoch, riak_object:syntactic_merge(LocalObj, IncomingObject)}.
+    %% case maybe_new_epoch_actor(VId, Epoch, LocalObj, IncomingObject) of
+    %%     {N, A} when N > Epoch ->  %% i.e. a new epoch is needed
+    %%         %% locally there is no entry for VId but the incoming
+    %%         %% object has an entry
+    %%         O2 = riak_object:syntactic_merge(LocalObj, IncomingObject),
+    %%         {N, riak_object:new_actor_epoch(O2, A)};
+    %%     {Epoch, _A} ->
+    %%         {Epoch, riak_object:syntactic_merge(LocalObj, IncomingObject)}
+    %% end.
+
+%% like vclock:timestamp() but monotinically increasing
+timestamp() ->
```

Comment:
It's my mistake, I was meant to do something smarter but never finished. Will address. So far the test has worked, but that might be good luck.